### DISCUSSION TITLE
Frontend TDD - character selection and difficulty setup

### DISCUSSION THREAD

**Discussion by KalpasWang** [Game-as-a-Service/Pandemic-Reign-of-Cthulhu#43] [General]
@wingtkw @sandytang886 
我比賽結束了，我想開始先實作出前端的 walking skeleton，我們要不要在前端用 TDD 開發呢？之前水球有說可以先用 cypress 做 e2e testing，再做出需要的功能。cypress 我雖然用過但是不熟，可以趁機學起來，不知道大家覺得如何？

**Comment by wingtkw** [comment]
可以
但覺得我們先develop 一些feature
再去弄測試會比較好
現在弄的大多屬於components ui
沒有太多functionality

**Comment by wingtkw** [comment]
目前pending 的functionality

選卡流程
- 等待選卡狀態
- 選角色卡
- 所有人選完卡後，進入選擇困難程度階段

選擇困難程度
如何呈現玩家擁有的卡

讓玩家看教程
呈現隊友狀態

地圖玩家、邪教徒、修格斯的呈現
地圖地點onclick 的執行行動

狀態儲存

**Reply by KalpasWang** [reply]
了解，那我來嘗試做選卡流程
